autogpt/config/config.py

```diff
@@ -41,7 +41,24 @@ def __init__(self) -> None:
         self.fast_llm_model = os.getenv("FAST_LLM_MODEL", "gpt-3.5-turbo")
         self.smart_llm_model = os.getenv("SMART_LLM_MODEL", "gpt-4")
         self.fast_token_limit = int(os.getenv("FAST_TOKEN_LIMIT", 4000))
-        self.smart_token_limit = int(os.getenv("SMART_TOKEN_LIMIT", 8000))
+
+        # It's better to set max limits and be explicit for better user experience IMO
+        model_limits = {
+            "gpt-4-32k": 32000,
+            "gpt-4": 8000,
+            "gpt-3.5-turbo": 4000,
+            "gpt-3.5-turbo-0301": 4000,
+        }
+
+        smart_token_limit = int(os.getenv("SMART_TOKEN_LIMIT", 4000))
```

Comment:
Why the default change from 8k to 4k?

Reply:
If for whatever reason it can't figure out what key it's on, 4000 seemed the safest. I don't mind changing it to 8000 and thinking more forward (assuming each model comes with an increase in token size?)

Reply:
Look at the line you deleted and replaced with this one, it used to be 8k. The smart llm by default is 4.0 so it should be 8k imo. https://help.openai.com/en/articles/7127966-what-is-the-difference-between-the-gpt-4-models

Reply:
That's a fair point, Ill change it

Reply:
[2ad9647](https://github.com/Significant-Gravitas/Auto-GPT/pull/4128/commits/2ad96477c273db109eba59397873f47407f722f2)